src/python/pants/backend/helm/goals/lint.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+
+from pants.backend.helm.subsystems.helm import HelmSubsystem
+from pants.backend.helm.target_types import HelmChartFieldSet, HelmChartLintStrictField
+from pants.backend.helm.util_rules.chart import HelmChart
+from pants.backend.helm.util_rules.tool import HelmProcess
+from pants.build_graph.address import Address
+from pants.core.goals.lint import LintResult, LintResults, LintTargetsRequest
+from pants.engine.process import FallibleProcessResult
+from pants.engine.rules import Get, MultiGet, collect_rules, rule
+from pants.engine.target import WrappedTarget
+from pants.engine.unions import UnionRule
+from pants.util.logging import LogLevel
+from pants.util.strutil import pluralize
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class HelmLintFieldSet(HelmChartFieldSet):
+    lint_strict: HelmChartLintStrictField
+
+
+class HelmLintRequest(LintTargetsRequest):
+    field_set_type = HelmLintFieldSet
+    name = HelmSubsystem.options_scope
+
+
+@rule(desc="Lint Helm charts", level=LogLevel.DEBUG)
+async def run_helm_lint(request: HelmLintRequest, helm_subsystem: HelmSubsystem) -> LintResults:
+    chart_targets = await MultiGet(
+        Get(WrappedTarget, Address, field_set.address) for field_set in request.field_sets
+    )
+    charts = await MultiGet(
+        Get(HelmChart, HelmChartFieldSet, HelmChartFieldSet.create(wrapped.target))
+        for wrapped in chart_targets
+    )
```

Comment:
I shouldn't need to use the `WrappedTarget` rule to perform this, as a `HelmLintFieldSet` is itself a `HelmChartFieldSet`.

However when I attempt to do:

```python
charts = await MultiGet(
  Get(HelmChart, HelmChartFieldSet, field_set)
  for field_set in request.field_sets
)
```

I get a type error as `field_set` is not exactly a `HelmChartFieldSet`. Which makes me think that `Get` is invaliant or basically that I don't know enough Python.

Reply:
Noting to do with Python actually since `Get` is a concept of the Pants engine, not Python generally. The Pants engine is type invariant (except with union types). 

See https://www.pantsbuild.org/docs/rules-api-concepts#valid-types

> Unlike Python in general, the engine uses exact type matches, rather than considering inheritance; even if Truck subclasses Vehicle, the engine will view these types as completely separate when deciding which rules to use.

Reply:
forgot to review that part of the docs... then I think that to avoid the usage of the `WrappedTarget`, I will need to change the request type of the rule, something like:

```python
@dataclass(frozen=True)
class HelmChartRequest:
  field_set: HelmChartFieldSet

@rule
async def get_helm_chart(request: HelmChartRequest) -> HelmChart:
 ...
```

And with that I could avoid the `WrappedTarget` with something like:

```python
charts = await MultiGet(
  Get(HelmChart, HelmChartRequest(field_set))
  for field_set in request.field_sets
```